src/script/event/WebSocketService.js

```diff
@@ -116,14 +114,15 @@ z.event.WebSocketService = class WebSocketService {
       };
 
       this.socket.onmessage = event => {
-        if (this._pingHasExperiencedSuspiciousInactivity()) {
-          const secondsSinceLastPing = (Date.now() - this.lastPingTime) / z.util.TimeUtil.UNITS_IN_MILLIS.SECOND;
-          this.logger.warn(`Message received but ping was inactive for "${secondsSinceLastPing}" sec, reconnecting.`);
-          return this.reconnect(WebSocketService.CHANGE_TRIGGER.LONG_INACTIVITY);
-        }
```

Comment:
The only downside of this flag-based approach that I can see is that now it's impossible to detect that `pingInterval` has not been started at all, previously based on Dates we could see that PINGs were not even being sent, now we can't. I don't immediately see how this situation can occur, just pointing out the difference 🙂

Reply:
Great. Thanks for taking the time rewriting it. The code is easier to read now and removes more lines than it adds. I like. 😺

Reply:
I think you can see that pings are not send when [inspecting the WebSocket](https://kaazing.com/inspecting-websocket-traffic-with-chrome-developer-tools/).